src/main/PersistentState.cpp

```diff
@@ -135,17 +160,39 @@ PersistentState::getState(PersistentState::Entry entry, SessionWrapper& session)
     ZoneScoped;
     releaseAssert(threadIsMain() ||
                   mApp.threadIsType(Application::ThreadType::APPLY));
-    return getFromDb(getStoreStateName(entry), session, getDBForEntry(entry));
+    return getFromDb(getStoreStateName(entry), session,
+                     getTableForEntry(entry));
 }
 
 void
-PersistentState::setState(PersistentState::Entry entry,
-                          std::string const& value, SessionWrapper& session)
+PersistentState::setMainState(PersistentState::Entry entry,
+                              std::string const& value, SessionWrapper& session)
 {
     ZoneScoped;
     releaseAssert(threadIsMain() ||
                   mApp.threadIsType(Application::ThreadType::APPLY));
-    updateDb(getStoreStateName(entry), value, session, getDBForEntry(entry));
+    if (entry > kRebuildLedger)
```

Comment:
nit: Now that we have `kLastEntryMain`, I would suggest using it everywhere instead of `kRebuildLedger`. It might also be useful to make `kLastEntryMain` to have the same value as `kRebuildLedger`, or name it `kMainEntryCount`. Either option would result in more clear inclusive/exclusive ranges, as currently the name suggests that it's an entry id, but in fact represents the entry count. 

The same goes for `kLastEntry`.